File: icure-sdk/src/commonMain/kotlin/com/icure/sdk/model/embed/FlatRateType.kt
package com.icure.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
public enum class FlatRateType() {
  @SerialName("physician")
  Physician,
  @SerialName("physiotherapist")
  Physiotherapist,
  @SerialName("nurse")
  Nurse,
  @SerialName("ptd")
  Ptd,
}

File: python-wrapper/src/python/model/embed/FlatRateType.py
from typing import Union, Dict
from enum import Enum

class FlatRateType(Enum):
	Physician = "physician"
	Physiotherapist = "physiotherapist"
	Nurse = "nurse"
	Ptd = "ptd"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'FlatRateType':
		if data == "physician":
			return FlatRateType.Physician
		elif data == "physiotherapist":
			return FlatRateType.Physiotherapist
		elif data == "nurse":
			return FlatRateType.Nurse
		elif data == "ptd":
			return FlatRateType.Ptd
		else:
			raise Exception(f"{data} is not a valid value for FlatRateType enum.")

__all__ = ['FlatRateType']